src/passes/librariesConverter.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  ContractDefinition,
+  ContractKind,
+  FunctionDefinition,
+  FunctionVisibility,
+} from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { ASTMapper } from '../ast/mapper';
+import { cloneASTNode } from '../utils/cloning';
+import { LIBRARY_CONTRACT_PREFIX } from '../utils/nameModifiers';
+import { getContainingSourceUnit } from '../utils/utils';
+
+// Clones a Library definition, converts it to regular Contract
+// and inserts it as a new child node of the library's parent.
+// This is made for transpiling the library as a contract.
+export class LibrariesConverter extends ASTMapper {
+  // Function to add passes that should have been run before this pass
+  addInitialPassPrerequisites(): void {
+    const passKeys: Set<string> = new Set<string>([]);
+    passKeys.forEach((key) => this.addPassPrerequisite(key));
+  }
+
+  visitContractDefinition(node: ContractDefinition, ast: AST): void {
+    if (node.kind === ContractKind.Library) {
+      const contractNode = cloneASTNode(node, ast);
+      contractNode.name = `${LIBRARY_CONTRACT_PREFIX}${node.name}`;
+      contractNode.kind = ContractKind.Contract;
+      this.externalizeFunctions(contractNode);
+      getContainingSourceUnit(node).appendChild(contractNode);
+    }
+  }
+
+  externalizeFunctions(node: ContractDefinition): void {
+    node.children.forEach((ch) => {
+      if (
+        ch instanceof FunctionDefinition &&
+        (ch.visibility === FunctionVisibility.Internal ||
+          ch.visibility === FunctionVisibility.Private)
+      ) {
+        ch.visibility = FunctionVisibility.External;
+      }
+    });
+  }
+}
```

Comment:
I don't understand why you're changing the new contract's visibility functions. I think they are meant to be private; by externalizing them security flaws can be created on the transpiled contract

Reply:
If I keep those functions private or internal, somewhere in the passes they get removed from the contract. As this contract is a cloned node, its components are not referenced by other nodes. I am not sure about what security flaws could be created.

Also, if I want to keep those functions internal or private, what can I do for they don't get removed?

Reply:
Function pruner removes all functions not visible from the outside. If this functions are private and internal and cannot be call by any form from an external function (directly or indirectly) then they are deleted. If you want them not to be deleted you need to design an external function which calls them.